frontend/sozisel-app/src/hooks/useFetchEventLiveResult.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  EventResultSubmittedSubscription,
+  QuizParticipantSummary,
+  useEventResultSubmittedSubscription,
+  useQuizParticipantsSummaryQuery,
+} from "../graphql";
+import { useEffect, useState } from "react";
+
+type QuizParticipantsSummary = ({
+  __typename?: "QuizParticipantSummary" | undefined;
+} & Pick<QuizParticipantSummary, "numberOfPoints" | "fullName">)[];
+
+interface EventResult {
+  pointSum: number;
+  completedTrialsNumber: number;
+  scoresDistribution: ScoreDistribution[];
+}
+
+interface ScoreDistribution {
+  score: number;
+  counter: number;
+}
+
+type Typename = "QuizSimpleResult";
+
+const addNewScore = (
+  scoreDistribution: ScoreDistribution[],
+  eventResult: EventResultSubmittedSubscription
+) => {
+  const resultData = eventResult.eventResultSubmitted?.resultData;
+  if (resultData?.__typename === "QuizSimpleResult") {
+    if (
+      Array.isArray(scoreDistribution) &&
+      scoreDistribution.filter((item) => item.score === resultData.totalPoints)
+        .length
+    ) {
+      return scoreDistribution.map((item) =>
+        item.score === resultData.totalPoints
+          ? { ...item, counter: (item.counter += 1) }
+          : item
+      );
+    }
+    return [
+      ...scoreDistribution,
+      { score: resultData.totalPoints, counter: 1 },
+    ];
+  }
+  return [];
+};
+
+const calculateNewPointSum = (
+  eventResults: EventResult,
+  eventResult: EventResultSubmittedSubscription,
+  typename: Typename
+): number => {
+  const resultData = eventResult.eventResultSubmitted?.resultData;
+  if (resultData?.__typename === typename) {
+    return eventResults.pointSum + resultData.totalPoints;
+  }
+  return 0;
+};
+
+const calcualteScoreDistribution = (
+  quizParticipantsSummary: QuizParticipantsSummary
+) => {
+  const uniqueScores = new Set(
+    quizParticipantsSummary.map((item) => item.numberOfPoints)
+  );
+  return Array.from(uniqueScores).map((score) => ({
+    score,
+    counter: quizParticipantsSummary.filter(
+      (item) => item.numberOfPoints === score
+    ).length,
+  }));
+};
+
+const useFetchEventLiveResult = (
```

Comment:
If you really want to create such complicated hooks and least put some comment what it does so other people does not have to deep dive into the code

Reply:
And from what I see this code is not generic whatsoever